src/drivers/intel/dmic.c

```diff
@@ -1071,7 +1071,11 @@ static int dmic_get_hw_params(struct dai *dai,
 
 	switch (dmic_prm[di]->num_pdm_active) {
 	case 1:
-		params->channels = 2;
+		params->channels =
+			dmic_prm[di]->pdm[0].enable_mic_a +
+			dmic_prm[di]->pdm[0].enable_mic_b +
+			dmic_prm[di]->pdm[1].enable_mic_a +
+			dmic_prm[di]->pdm[1].enable_mic_b;
```

Comment:
I don't get the commit message, with this change the values could be 1..4, no?

Reply:
maybe even 0 to 4. 0 is no flags are set where which we should check and return error with trace.

Reply:
Yes, 0 could be possible if not using the above topology macros but direct byte values in topology, I need to check that it's returned error before the code can reach this.  3 or 4 can happen only with active controllers value 2. The IPC is like this due to move into fixed length struct and not being able to clean up legacy without major ABI bump.